shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2272,11 +2291,62 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   } else if ([method isEqualToString:kUpdateConfigMethod]) {
     [self updateConfig:args];
     result(nil);
+  } else if ([method isEqualToString:kOnInteractiveKeyboardPointerMoveMethod]) {
+    CGFloat pointerY = (CGFloat)[args[@"pointerY"] doubleValue];
+    [self handlePointerMove:pointerY];
+    result(nil);
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (void)handlePointerMove:(CGFloat)pointerY {
+  double screenHeight = [UIScreen mainScreen].bounds.size.height;
+  double keyboardHeight = _keyboardRect.size.height;
+  // Determines if pointer goes overtop of the keyboard.
+  if (screenHeight - keyboardHeight <= pointerY) {
+    // If no screenshot has been taken.
+    if (_keyboardView.superview != nil) {
+      [self takeScreenshot];
+      [self hideKeyboardWithoutAnimation];
+    } else {
+      [self setKeyboardContainerHeight:pointerY];
+    }
+  } else {
+    // If pointer is not overtop of the original keyboard keep screenshot in place.
+    if (_keyboardView.superview != nil) {
+      _keyboardViewContainer.frame = _keyboardRect;
+    }
+  }
+}
+
+- (void)setKeyboardContainerHeight:(CGFloat)pointerY {
+  CGRect frameRect = _keyboardRect;
+  frameRect.origin.y = pointerY;
+  _keyboardViewContainer.frame = frameRect;
+}
+
+- (void)hideKeyboardWithoutAnimation {
+  [UIView setAnimationsEnabled:NO];
+  _firstResponder = UIApplication.sharedApplication.keyWindow.flt_firstResponder;
+  [_firstResponder resignFirstResponder];
+  [UIView setAnimationsEnabled:YES];
+}
+
+- (void)takeScreenshot {
```

Comment:
Consider renaming to something more specific for readability -- e.g. `take KeyboardSnapshot`

Reply:
Definitely, I also added add to screen to be even better for readability.